cypress/integration/ui/archiving-deal-group.js

```diff
@@ -0,0 +1,195 @@
+      cy.wait(1000).get('input[type="search"]').clear().type(dealGroupName);
+      if (dealGroupsCount === 1) {
+        cy.get('[data-qa="deal-group-dashboard--select-deal-group"]').first().click().then(() => {
+          cy.get('header').should('contain', dealGroupName);
+        });
+        ArchivedDealGroupName = dealGroupRows[0].name;
+      } else {
+        cy.log('There is no/more than one deal group');
+      }
+      cy.get('[data-qa="deal-group-detail--archive_button"]').click().wait(1500);
+      cy.get('deal-group-detail > message-modal button.button.button--primary').click().wait(1800);
+
+      cy.get('[data-qa="deal-dashboard--add-deal"]').should('be.disabled').wait(500);
+      cy.get('.icon-archive').should('exist').wait(1000);
+      cy.get('[data-qa="deal-group-detail--archive_button"]').should('contain', 'Unarchive');
+      cy.get('.breadcrumb').click().wait(200); // User return to deal groups landing page
+      if (dealGroupsCount === 0) {
+        cy.wait(1000).get('input[type="search"]').clear().type(dealGroupName)
+          .wait(1500);
+      } else {
+        cy.wait(1000).get('input[type="search"]').clear().type(ArchivedDealGroupName)
+          .wait(1500);
+      }
+      cy.get('[data-qa="deal-group-dashboard--exclude-tests"]').click().wait(1000);
+      cy.get('.t-regular').should('exist').wait(1000); // No deal group is returned
+      cy.get('div.dropdown-toggle').click().wait(1000);
+      cy.get('ul li:nth-child(3) a').click().wait(1000); // Clicking on archive option
+      cy.get('[data-qa="deal-group-dashboard--select-deal-group"]').should('exist');
+      cy.get('.t-regular').should('not.exist').wait(1000); // "No results found" text should NOT display
+      cy.get('div.table data-row:nth-child(2) p2:nth-child(1)').should('contain', 'ARCHIVED').wait(1000);
+    });
+
+    it('Clicking on "Unrchive" button in the deal group entry page', () => {
+      cy.get('[data-qa="deal-group-dashboard--select-deal-group"]').click().wait(2000);
+      cy.get('[data-qa="deal-group-detail--archive_button"]').click().wait(1500);
+      cy.get('deal-group-detail > message-modal button.button.button--primary').click().wait(2000);
+    });
+
+    it('Verifying that deal group is Unarchived', () => {
+      cy.get('[data-qa="deal-dashboard--add-deal"]').should('be.enabled').wait(500);
+      cy.get('.icon-archive').should('not.exist').wait(1000);
+      cy.get('[data-qa="deal-group-detail--archive_button"]').should('contain', 'Archive');
+      cy.get('.breadcrumb').click().wait(200); // User return to deal groups landing page
+      if (dealGroupsCount === 0) {
+        cy.wait(1000).get('input[type="search"]').clear().type(dealGroupName)
+          .wait(1500);
+      } else {
+        cy.wait(1000).get('input[type="search"]').clear().type(ArchivedDealGroupName)
+          .wait(1500);
+      }
+      cy.get('[data-qa="deal-group-dashboard--exclude-tests"]').click().wait(1000);
```

Comment:
This does not verify that the deal-group that was recently archived, appears.  It just look for anything to appear